authz: add docs and links to background permissions syncing

unknwon opened the pull request:
Adds a section in docs to explain "complete sync vs incremental sync", and adds links to docs in corresponding pages.

Fixes #10190

ryan-blunden commented:
If I understand this correctly, a "complete sync" is only done once at the start, either when a repository or user is first added? Then an incremental sync is just reflecting any changes in permissions that have happened since then to bring Sourcegraph up-to-date with the code host?

If that' the case, I would try adjusting the content here to explain the sequence of when each type of sync happens, and in the case of an incremental sync, what would trigger an incremental sync?

unknwon commented:
> If I understand this correctly, a "complete sync" is only done once at the start, either when a repository or user is first added? Then an incremental sync is just reflecting any changes in permissions that have happened since then to bring Sourcegraph up-to-date with the code host?

Hmm... a "complete sync" is what happens periodically, which has side effect of triggering many incremental syncs to reflect any permissions changes since last "complete sync" of those affected repositories or users.

unknwon commented:
Merging as-is, happy to address any post-merge comments!
